src/applications/bmqstoragetool/m_bmqstoragetool_compositesequencenumber.h

```diff
@@ -0,0 +1,218 @@
+    /// filled with error description.
+    CompositeSequenceNumber& fromString(bsl::ostream&      errorDescription,
+                                        const bsl::string& seqNumString);
+
+    // ACCESSORS
+
+    /// Return `true` if the value of this object is not set.
+    bool isUnset() const;
+
+    /// Return Primary Lease Id value.
+    unsigned int leaseId() const;
+
+    /// Return Sequence Number value.
+    bsls::Types::Uint64 sequenceNumber() const;
+
+    /// Write the value of this object to the specified output `stream` in a
+    /// human-readable format, and return a reference to `stream`.
+    /// Optionally specify an initial indentation `level`.  If `level` is
+    /// specified, optionally specify `spacesPerLevel`, whose absolute value
+    /// indicates the number of spaces per indentation level for this
+    /// object.  If `level` is negative, suppress indentation of the first
+    /// line.  If `spacesPerLevel` is negative, format the entire output on
+    /// one line, suppressing all but the initial indentation (as governed
+    /// by `level`).  If `stream` is not valid on entry, this operation has
+    /// no effect.  Note that this human-readable format is not fully
+    /// specified, and can change without notice.
+    bsl::ostream&
+    print(bsl::ostream& stream, int level = 0, int spacesPerLevel = 4) const;
+};
+
+// FREE OPERATORS
+
+// -----------------------------
+// class CompositeSequenceNumber
+// -----------------------------
+
+/// Write the value of the specified `rhs` object to the specified output
+/// `stream` in a human-readable format, and return a reference to `stream`.
+/// Note that this human-readable format is not fully specified, and can
+/// change without notice.
+bsl::ostream& operator<<(bsl::ostream&                  stream,
+                         const CompositeSequenceNumber& rhs);
+
+/// Return true if the specified `lhs` instance is equal to the
+/// specified `rhs` instance, false otherwise.
+bool operator==(const CompositeSequenceNumber& lhs,
+                const CompositeSequenceNumber& rhs);
+
+/// Return true if the specified `lhs` instance is less than the
+/// specified `rhs` instance, false otherwise.
+bool operator<(const CompositeSequenceNumber& lhs,
+               const CompositeSequenceNumber& rhs);
+
+/// Return true if the specified `lhs` instance is less or equal to the
+/// specified `rhs` instance, false otherwise.
+bool operator<=(const CompositeSequenceNumber& lhs,
+                const CompositeSequenceNumber& rhs);
+
+// ============================================================================
+//                             INLINE DEFINITIONS
+// ============================================================================
+
+// =============================
+// class CompositeSequenceNumber
+// =============================
+
+// ACCESSORS
+
+inline bool CompositeSequenceNumber::isUnset() const
+{
+    return d_isUnset;
```

Comment:
There are expressions like `if (!isUnset()) {...}` everywhere. Double negation looks overcomplicated to me. And furthermore nobody really "unsets" this object)). `IsSet` or `IsInitialized` would be much more readable.